### PR TITLE
Add isconst api to problem.

### PR DESCRIPTION
**Pull request by schneith**

### CODE REVIEW

File: is_const.patch [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+ 
++bool ProblemImpl::IsParameterBlockConstant(double* values) const {
++  const ParameterBlock* parameter_block =
++      FindWithDefault(parameter_block_map_, values, NULL);
++  if (parameter_block == NULL) {
++    LOG(FATAL) << "Parameter block not found: " << values
++               << ". You must add the parameter block to the problem before "
++               << "it can be queried.";
++  }
++  parameter_block->IsConstant();
```
Comment (tcies) [inline is_const.patch]: return? :)

**Comment by tcies** [comment]
Quick question: Do you think it would be possible to have a separate package for a tampered-with ceres? So that "normal" ceres users could just upgrade ceres without having to go through your patches? Haven't though that through though...

**Comment by tcies** [comment]
ocoqolgtm

File: is_const.patch (hunk 1 above)
Comment (schneith) [inline is_const.patch]: oh, that might make sense :) good catch

**Comment by schneith** [comment]
I will probably not need this patch, if we can export the internal headers with the other patch here https://github.com/ethz-asl/ceres_catkin/pull/28

**Pull request merged**